<!-- ros launch file for 3f gripper onn modbus tcp -->
<!-- launch with: roslaunch uclv_robotiq 3f_gripper.launch -->

<launch>
    <arg name="modbus_type" default="tcp"/> <!-- tcp, rtu, or rtu_over_tcp -->
    <arg name="ip_address" default="192.168.1.11"/>
    <arg name="serial_port" default="/dev/ttyUSB0"/>
    <arg name="port" default="502"/>
    <arg name="slave_id" default="9"/>
    <arg name="baudrate" default="115200"/>
    <arg name="gripper_name" default="3f_gripper"/>

    <!-- launch gripper node -->
    <node name="gripper_node" pkg="uclv_robotiq_ros" type="robotiq_3f_gripper_node" ns="$(arg gripper_name)" output="screen">
        <param name="modbus_type" value="$(arg modbus_type)"/>
        <param name="ip_address" value="$(arg ip_address)"/>
        <param name="serial_port" value="$(arg serial_port)"/>
        <param name="baudrate" value="$(arg baudrate)"/>
        <param name="port" value="$(arg port)"/>
        <param name="slave_id" value="$(arg slave_id)"/>
    </node>
</launch>

